runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -958,6 +959,11 @@ private Debuggee registerDebuggee(CloudDebugger debuggerClient, String uniquifie
         } catch (InvalidProtocolBufferException e) {
           throw new RuntimeException("Error parsing artifact file payload.", e);
         }
+        if (stagedPaths.contains(filePayload.getPath())) {
```

Comment:
I think we don't need this with the new artifact container ? (we do not need to support multiple environments for older Java versions that do not support the new artifact container).

Reply:
Heejong, can you confirm this ? If so we might not need this PR.

Reply:
I think the new artifact service should fix the issue. But we might still want to merge this PR since the duplication is not right in the first place when we create a list of artifacts from all environments. Also, there are some cases where the new artifact service can't cover e.g. customized user agent string.

Reply:
Wouldn't this prevent different environments from having the same dependency ?

Also, please add unit tests.

Reply:
The different environments still can have the same dependency. It only stops staging the same artifact multiple times so that a list of staged packages doesn't have any duplicates.

Added a test.